velox/expression/CastExpr-inl.h

```diff
@@ -327,37 +340,44 @@ VectorPtr CastExpr::applyDecimalToIntegralCast(
   const auto precisionScale = getDecimalPrecisionScale(*fromType);
   const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
   const auto scaleFactor = DecimalUtil::kPowersOfTen[precisionScale.second];
-  const auto castToIntByTruncate =
-      context.execCtx()->queryCtx()->queryConfig().isCastToIntByTruncate();
-  applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
-    auto value = simpleInput->valueAt(row);
-    auto integralPart = value / scaleFactor;
-    if (!castToIntByTruncate) {
+  if (hooks_->truncate()) {
+    rows.applyToSelected([&](auto row) {
+      const auto integralPart = simpleInput->valueAt(row) / scaleFactor;
+      if (integralPart > std::numeric_limits<To>::max() ||
+          integralPart < std::numeric_limits<To>::min()) {
+        resultBuffer[row] = To(integralPart);
+        return;
+      }
+      resultBuffer[row] = static_cast<To>(integralPart);
```

Comment:
Curious what different behavior between To() and static_cast<To>() is utilized here?

Reply:
Thanks for your review! They behaves the same for overflow case, so the branch to handle integer overflow can be removed.